visualization/app/codeCharta/util/debounce.ts

```diff
@@ -0,0 +1,7 @@
+export const debounce = <F extends (...arguments_: Parameters<F>) => ReturnType<F>>(f: F, waitInMS = 300) => {
```

Comment:
Let's keep the former scope by using a regular function, accessing its scope, saving it to a variable and then binding the passed through function to that scope.
Also please use at least fn or func as variable name.  Single character variable names are always tricky.

Reply:
I have a strong dislike against any usage of an implicit "this". But it is heavily opinionated, so I just added it.

Reply:
(no need to save `const self = this` as an arrow function  keeps outer this)

Reply:
I am not certain I can follow about the implicit context. It's now explicitly set to the original context? :-)

Reply:
We can have a nice :coffee: / :beers: about it the next time we see each other :)